rest/blip_client_test.go

```diff
@@ -0,0 +1,205 @@
+			}
+		}
+
+		if !request.NoReply() {
+			response := request.Response()
+			if btc.Deltas {
+				// Enable deltas from the client side
+				response.Properties["deltas"] = "true"
+			}
+
+			b, err := json.Marshal(wantedRevs)
+			if err != nil {
+				panic(err)
+			}
+
+			response.SetBody(b)
+			btc.Responses <- response
+		}
+	}
+
+	bt.blipContext.HandlerForProfile[messageRev] = func(request *blip.Message) {
+		btc.Requests <- request
+		log.Printf("client got rev request: %v", request)
+
+		docID := request.Properties["id"]
+		revID := request.Properties["rev"]
+
+		body, err := request.Body()
+		if err != nil {
+			panic(err)
+		}
+
+		// If deltas are enabled, and we see a deltaSrc property, we'll need to patch it before storing
+		if btc.Deltas && bytes.Contains(body, []byte(`"deltaSrc":`)) {
+			log.Printf("    recieved delta: %s %s %s", docID, revID, body)
+
+			// unmarshal body to extract deltaSrc
+			var delta map[string]interface{}
+			if err := json.Unmarshal(body, &delta); err != nil {
+				panic(err)
+			}
+
+			var old map[string]interface{}
+			if deltaSrc, ok := delta["deltaSrc"]; ok {
+				btc.docsLock.RLock()
+				oldBytes := btc.docs[docID][deltaSrc.(string)]
+				btc.docsLock.RUnlock()
+				if err := json.Unmarshal(oldBytes, &old); err != nil {
+					panic(err)
+				}
+			}
+
+			if err := fleecedelta.PatchJSON(&old, fleecedelta.Delta(body)); err != nil {
+				panic(err)
+			}
+			delete(old, "deltaSrc") // remove deltaSrc after patching
+
+			body, err = json.Marshal(old)
+			if err != nil {
+				panic(err)
+			}
+		}
+
+		btc.docsLock.Lock()
+		btc.docs[docID] = map[string][]byte{revID: body}
+		btc.docsLock.Unlock()
+		log.Printf("    stored doc: %s %s %s", docID, revID, body)
+
+		if !request.NoReply() {
+			response := request.Response()
+			response.SetBody([]byte(`[]`))
+			btc.Responses <- response
+		}
+	}
+
+	bt.blipContext.HandlerForProfile[messageGetAttachment] = func(request *blip.Message) {
+		btc.Requests <- request
+		panic("getAttachment not implemented")
+	}
+
+	bt.blipContext.DefaultHandler = func(request *blip.Message) {
+		btc.Requests <- request
+		log.Printf("defaultHandler got request: %v", request)
+	}
+
+	return &btc, nil
+}
+
+// Start will begin a continous replication since 0 between the client and server, subscribibng to the * channel.
+func (btc *BlipTesterClient) Start() {
+	getCheckpointRequest := blip.NewRequest()
+	getCheckpointRequest.SetProfile(messageGetCheckpoint)
+	getCheckpointRequest.Properties[blipClient] = btc.id
+
+	if !btc.bt.sender.Send(getCheckpointRequest) {
+		log.Fatalf("unable to send getCheckpointRequest")
+	}
+
+	subChangesRequest := blip.NewRequest()
+	subChangesRequest.SetProfile(messageSubChanges)
+	subChangesRequest.Properties[subChangesContinuous] = "true"
+	subChangesRequest.Properties[subChangesSince] = "0"
+	subChangesRequest.Properties[subChangesFilter] = "sync_gateway/bychannel"
+	subChangesRequest.Properties[subChangesChannels] = "*"
```

Comment:
Can omit channel filter for the simple case (will replicate all channels the user has access to).  Could stub out a StartChannelFiltered() API?